cmd/peer-rest-server.go

```diff
@@ -1008,5 +1008,7 @@ func registerPeerRESTHandlers(router *mux.Router) {
 	subrouter.Methods(http.MethodPost).Path(SlashSeparator + peerRESTMethodBackgroundHealStatus).HandlerFunc(server.BackgroundHealStatusHandler)
 	subrouter.Methods(http.MethodPost).Path(SlashSeparator + peerRESTMethodLog).HandlerFunc(server.ConsoleLogHandler)
 
+	// Note: Update the preceding version range whenever the rpc version gets updated.
+	router.Methods(http.MethodPost).Path(minioReservedBucketPath + "/peer/" + "{v[1-4]}" + SlashSeparator + "{a-z}").HandlerFunc(httpTraceAll(versionMismatchHandler))
```

Comment:
Actually, this is not what I was thinking about how to fix this issue. But more we fix it in the `notFoundHandler()`  such that we are able to provide an error with more context like a full structure with the "API" call perhaps that was attempted etc. 

We can return a different set of calls but we can verify the resource there as a switch case it would work in a generic manner for all router registrations especially 

admin-router.go 
peer-rest-server.go
storage-rest-server.go
lock-rest-server.go

etc.

Reply:
The problem i went through was that the request was not reaching notFoundHandler at all. I tried this by dropping in some fmt.prints inside the handler. I could not see the request reaching the notFoundHandler. The error that we see is from http.Do() i think

Reply:
> The problem i went through was that the request was not reaching notFoundHandler at all. I tried this by dropping in some fmt.prints inside the handler. I could not see the request reaching the notFoundHandler. The error that we see is from http.Do() i think

It should since that is the handler which returns the error